README.md

```diff
@@ -43,6 +43,97 @@ let output = p.process('<template>Hi</template>');
 console.log(output);
 ```
 
+## API
+
+### `Preprocessor`
+
+All `content-tag` public API lives on the `Preprocessor` object.
+
+### `Preprocessor.process(src: string, filename?: string): string;`
+
+Parses a given source code string using the `content-tag` spec into standard
+JavaScript.
+
+```ts
+import { Preprocessor } from 'content-tag';
+let p = new Preprocessor();
+let output = p.process('<template>Hi</template>');
+```
+
+### `Preprocessor.parse(src: string, filename?: string): Parsed[];`
```

Comment:
I didn't document what adding the filename does bc I don't know the answer to that.

Reply:
It makes the error messages nicer if you get a parse error.

Reply:
Any reason not to pass this as an option to the generated `template()` so it ends up in the metadata of the ultimately compiled template?

Reply:
We need to do some design here. The old concept of template module names doesn't really map to template tag. 

Templates and files are no longer one to one. It would be easy to produce a very confusing component tree in the inspector for example where multiple layers of components all have "the same name" because they're defined in the same file. 

Classically the module names were ember's weird module namespace. Not real file names. If we use real file names they get unwieldy quickly and leak system information. Inventing a mapping requires design. 

I really think we need to address glimmer stack debugging in a holistic way. We need to map to your actual template, not just the file the template is in.

Reply:
I would add that content-tag is probably not the right layer to inject this even if it's desirable. content-tag is purely syntactic transformation. You still need template compilation (as implemented in babel-plugin-ember-template-compilation) afterward, and it's the template compiler that can enforce policy about inserting module names.